Remotion/Web/UnitTests/Core/ContentSecurityPolicy/CspEnabledHtmlTextWriterTests.cs

```diff
@@ -0,0 +1,205 @@
+    {
+      _writer.AddAttribute("style", "color:white;");
+      _writer.RenderBeginTag(HtmlTextWriterTag.Button);
+      _writer.RenderEndTag();
+
+      var document = _htmlHelper.GetResultDocument();
+      var element = _htmlHelper.GetAssertedChildElement(document, "button", 0);
+      _htmlHelper.AssertNoAttribute(element, "data-inline-event-target");
+
+      _clientScriptMock.Verify();
+    }
+
+    [Test]
+    public void AddAttribute_WithMoreThanOneAttribute_NonceValueIsRendered ()
+    {
+      _randomNumberGeneratorMock
+          .Setup(m => m.GenerateAlphaNumericNonce())
+          .Returns("eventTargetID")
+          .Verifiable();
+
+      _clientScriptMock
+          .Setup(m => m.RegisterStartupScriptBlock(_pageMock.Object, typeof(CspEnabledHtmlTextWriter), It.IsAny<string>(), It.IsAny<string>()))
+          .Verifiable();
+
+      _writer.AddAttribute("onclick", "console.info('test1');");
+      _writer.AddAttribute("onchange", "console.info('test2');");
+      _writer.RenderBeginTag(HtmlTextWriterTag.Button);
+      _writer.RenderEndTag();
+
+      var document = _htmlHelper.GetResultDocument();
+      var element = _htmlHelper.GetAssertedChildElement(document, "button", 0);
+      _htmlHelper.AssertAttribute(element, "data-inline-event-target", "eventTargetID");
+
+      _clientScriptMock.Verify();
+      _randomNumberGeneratorMock.Verify(m => m.GenerateAlphaNumericNonce(), Times.Once());
+    }
+
+    [Test]
+    public void AddAttribute_WithMoreThanOneTag_NonceValueIsRendered ()
+    {
+      _randomNumberGeneratorMock
+          .SetupSequence(m => m.GenerateAlphaNumericNonce())
+          .Returns("eventTargetID1")
+          .Returns("eventTargetID2");
+
+      _clientScriptMock
+          .Setup(m => m.RegisterStartupScriptBlock(_pageMock.Object, typeof(CspEnabledHtmlTextWriter), It.IsAny<string>(), It.IsAny<string>()))
+          .Verifiable();
+
+      _writer.RenderBeginTag("root");
+
+      _writer.AddAttribute("onchange", "console.info('test1');");
+      _writer.RenderBeginTag(HtmlTextWriterTag.Label);
+      _writer.RenderEndTag();
+
+      _writer.AddAttribute("onclick", "console.info('test2');");
+      _writer.RenderBeginTag(HtmlTextWriterTag.Button);
+      _writer.RenderEndTag();
+
+      _writer.RenderEndTag();
+
+      var document = _htmlHelper.GetResultDocument();
+      var root = _htmlHelper.GetAssertedChildElement(document, "root", 0);
+      _htmlHelper.AssertChildElementCount(root, 2);
+
+      var element1 = _htmlHelper.GetAssertedChildElement(root, "label", 0);
+      _htmlHelper.AssertAttribute(element1, "data-inline-event-target", "eventTargetID1");
+
+      var element2 = _htmlHelper.GetAssertedChildElement(root, "button", 1);
+      _htmlHelper.AssertAttribute(element2, "data-inline-event-target", "eventTargetID2");
+
+      _clientScriptMock.Verify();
+    }
+
+    [Test]
+    public void AddAttribute_WithNotSupportedEventType_ThrowsNotSupportedException ()
+    {
+      Assert.That(
```

Comment:
note: when changing the exception type, we have helper for assering argument exception messages to make it easy when handling parmaeter names.